### PR TITLE
Add new data of (Fast Bridge Federation P2SH, Derivation Arguments ha…

### PR DESCRIPTION
**Pull request by pamgonzalez**
…sh, Federation P2SH) in StorageProvider.

Save the data of (Fast Bridge Federation P2SH, Derivation Arguments hash, Federation P2SH) in Storage Provider
Returns an object which included the info of (Fast Bridge Federation P2SH, Federation P2SH) when a Derivation Arguments hash is given.

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/fastBridge/FastBridgeFederationP2SH.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package co.rsk.peg.fastBridge;
+
+import co.rsk.bitcoinj.core.Sha256Hash;
+
+public class FastBridgeFederationP2SH {
+    private byte[] fastBridgeFederationP2SH;
+    private byte[] federationP2SH;
```
Comment (josedahlquist) [inline rskj-core/src/main/java/co/rsk/peg/fastBridge/FastBridgeFederationP2SH.java:7]:
```suggestion
public class FastBridgeFederationInformation {
    private byte[] fastBridgeFederationScriptHash;
    private byte[] federationScriptHash;
```

**Review by guidohernan93 — APPROVED** [review]

File: rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java [hunk 2]
```diff
@@ -626,6 +630,49 @@ private void saveDerivationArgumentsScriptHash() {
...
+            return Optional.empty();
+        }
+
+        if (fastBridgeFederationP2SH == null || fastBridgeFederationP2SH.length == 0) {
+            return Optional.empty();
+        }
+
+        FastBridgeFederationInformation fastBridgeFederationInformation = this.safeGetFromRepository(
+                getStorageKeyForfastBridgeFederationInformation(fastBridgeFederationP2SH),
+                BridgeSerializationUtils::deserializeFederationBridge
```
Comment (guidohernan93) [inline rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java:644]: should call deserializeFastBridgeInformation method

File: rskj-core/src/main/java/co/rsk/peg/BridgeSerializationUtils.java [hunk 3]
```diff
@@ -743,6 +744,33 @@ public static CoinbaseInformation deserializeCoinbaseInformation(byte[] data) {
...
 
+    public static FastBridgeFederationInformation deserializeFastBridgeInformation(byte[] data) {
+        if ((data == null) || (data.length == 0)) {
+            throw new RuntimeException("Invalid serialized Fast Bridge Federation: null data or no data to deserialize");
+        }
+
+        RLPList rlpList = (RLPList)RLP.decode2(data).get(0);
+
+        if (rlpList.size() != 2) {
+            throw new RuntimeException(String.format("Invalid serialized Fast Bridge Federation: expected 2 value but got %n", rlpList.size()));
```
Comment (guidohernan93) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSerializationUtils.java:755]: `%n` should be `%d` I think

File: rskj-core/src/main/java/co/rsk/peg/BridgeSerializationUtils.java [hunk 4]
```diff
@@ -743,6 +744,33 @@ public static CoinbaseInformation deserializeCoinbaseInformation(byte[] data) {
         return RLP.encodeList(rlpElements);
     }
 
+    public static FastBridgeFederationInformation deserializeFastBridgeInformation(byte[] data) {
+        if ((data == null) || (data.length == 0)) {
+            throw new RuntimeException("Invalid serialized Fast Bridge Federation: null data or no data to deserialize");
+        }
+
+        RLPList rlpList = (RLPList)RLP.decode2(data).get(0);
```
Comment (guidohernan93) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSerializationUtils.java:752]: Should we check if `rlpList` is null?

Comment (pamgonzalez) [reply rskj-core/src/main/java/co/rsk/peg/BridgeSerializationUtils.java:752]: decode2 function can returns empty list or a list populates with values. If data is null, then return empty list.

**Review by guidohernan93 — APPROVED** [review]

File: rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java [hunk 5]
```diff
@@ -626,6 +630,49 @@ private void saveDerivationArgumentsScriptHash() {
...
+                BridgeSerializationUtils::deserializeFastBridgeInformation
+        );
+        if (fastBridgeFederationInformation == null) {
+            return Optional.empty();
+        }
+
+        return Optional.of(fastBridgeFederationInformation);
+    }
+
+    public void setFastBridgeFederationInformation(byte[] fastBridgeScriptHash, FastBridgeFederationInformation fastBridgeFedP2SH) {
```
Comment (josedahlquist) [inline rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java:653]:
```suggestion
    public void setFastBridgeFederationInformation(byte[] fastBridgeScriptHash, FastBridgeFederationInformation fastBridgeFederationInformation) {
```

**Review by josedahlquist — APPROVED** [review]
Just a minor correction in naming. LGTM

**Pull request merged**